src/main/java/ti4/generator/UnitRenderGenerator.java

```diff
@@ -0,0 +1,545 @@
+            (ButtonHelper.isLawInPlay(game, "articles_war") ||
+                ButtonHelper.isLawInPlay(game, "absol_articleswar"));
+    }
+
+    private boolean isWarsunWithSchematics(UnitKey unitKey) {
+        return unitKey.getUnitType() == UnitType.Warsun &&
+            ButtonHelper.isLawInPlay(game, "schematics");
+    }
+
+    private void drawMechTearDecal(UnitKey unitKey, int imageX, int imageY) {
+        String imagePath = "agenda_articles_of_war" + DrawingUtil.getBlackWhiteFileSuffix(unitKey.getColorID());
+        BufferedImage mechTearImage = ImageHelper.read(resourceHelper.getTokenFile(imagePath));
+        tileGraphics.drawImage(mechTearImage, imageX, imageY, null);
+    }
+
+    private void drawWarsunCrackDecal(UnitKey unitKey, int imageX, int imageY) {
+        String imagePath = "agenda_publicize_weapon_schematics" + DrawingUtil.getBlackWhiteFileSuffix(unitKey.getColorID());
+        BufferedImage wsCrackImage = ImageHelper.read(resourceHelper.getTokenFile(imagePath));
+        tileGraphics.drawImage(wsCrackImage, imageX, imageY, null);
+    }
+
+    private void drawUnitTags(UnitKey unitKey, Player player, ImagePosition imagePos, int iteration) {
+        if (iteration != 0 || UnitType.Infantry.equals(unitKey.getUnitType()) || !game.isShowUnitTags()) {
+            return;
+        }
+
+        UnitModel unitModel = game.getUnitFromUnitKey(unitKey);
+        if (player == null || unitModel == null || !unitModel.getIsShip()) {
+            return;
+        }
+
+        String factionTag = player.getFactionModel().getShortTag();
+        BufferedImage plaquette = ImageHelper.read(resourceHelper.getUnitFile("unittags_plaquette.png"));
+        Point plaquetteOffset = getUnitTagLocation(unitKey.asyncID());
+
+        int tagX = imagePos.x() + plaquetteOffset.x;
+        int tagY = imagePos.y() + plaquetteOffset.y;
+
+        // Draw plaquette background
+        tileGraphics.drawImage(plaquette, tagX, tagY, null);
+
+        // Draw faction icon
+        DrawingUtil.drawPlayerFactionIconImage(tileGraphics, player, tagX, tagY, 32, 32);
+
+        // Draw faction tag text
+        tileGraphics.setColor(Color.WHITE);
+        Rectangle textBounds = new Rectangle(tagX + 25, tagY + 17, 40, 13);
+        DrawingUtil.drawCenteredString(tileGraphics, factionTag, textBounds, Storage.getFont13());
+    }
+
+    private SystemContext buildSystemContext(Tile tile, UnitHolder unitHolder, Player frogPlayer) {
+        boolean isSpace = unitHolder.getName().equals(Constants.SPACE);
+        boolean isMirage = unitHolder.getName().equals(Constants.MIRAGE);
+        boolean hasMirage = false;
+        if (isSpace) {
+            Set<String> tokenList = unitHolder.getTokenList();
+            hasMirage = tokenList.stream().anyMatch(tok -> tok.contains("mirage"))
+                && (tile.getPlanetUnitHolders().size() != 3 + 1);
+        }
+
+        Map<String, String> jailTiles = Map.of(
```

Comment:
We can remove the "jailTiles" handling - these are deprecated and have been replaced the NomBox in the player area tied to the /capture commands